### PR TITLE
Diagnostics app

### PR DESCRIPTION
**Pull request by benitav**
Initial release of tools/diagnostics-app

### CODE REVIEW

File: demos/diagnostics-app/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Diagnostics app
+
+This diagnostics app presents data from the perspective of a specific user (no server-side stats) and can be used to:
+- See stats about the data in a user's local database.
+- Inspect a user's sync buckets.
```
Comment (benitav) [inline demos/diagnostics-app/README.md:5]:
```suggestion
- Inspect tables and sync buckets on the device.
```

File: demos/diagnostics-app/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# Diagnostics app
+
+This diagnostics app presents data from the perspective of a specific user (no server-side stats) and can be used to:
+- See stats about the user's local database.
+- Inspect tables and sync buckets on the device.
+- Identify common issues.
```
Comment (benitav) [inline demos/diagnostics-app/README.md:6]:
```suggestion
- Identify common issues. E.g. too many sync buckets.
```

**Comment by DominicGBauer** [comment]
Should the diagnostics app not be in it's own folder instead of `demos`?

**Comment by benitav** [comment]
> Should the diagnostics app not be in its own folder instead of `demos`?

Are you thinking an `apps` folder to distinguish examples/demos from apps that developers can use? If so, I don't feel strongly about it at this point, since we currently just have one of these. If we publish more it might make sense.

**Comment by rkistner** [comment]
> Should the diagnostics app not be in its own folder instead of `demos`?

Also happy for other suggestions - I just placed it there since it followed the same structure as other demos.

**Comment by DominicGBauer** [comment]
I would say `tools` instead of `apps` since tools are probably the only apps we will make.

File: packages/common/src/db/schema/Schema.ts [hunk 3]
```diff
@@ -16,7 +16,9 @@ export class Schema<S extends SchemaType = SchemaType> {
   */
   readonly types: SchemaTableType<S>;
   readonly props: S;
-  constructor(public tables: ClassicTable[] | S) {
+  readonly tables: ClassicTable[];
```
Comment (stevensJourney) [inline packages/common/src/db/schema/Schema.ts:19]: We should add a changeset for this update for it to be published.

File: demos/diagnostics-app/src/app/views/layout.tsx [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    return () => l();
+  }, []);
+  const drawerWidth = 320;
+
+  const drawer = (
+    <>
+      <S.PowerSyncLogo alt="PowerSync Logo" width={250} height={100} src="/powersync-logo.svg" />
+      <Divider />
+      <List>
+        {NAVIGATION_ITEMS.map((item) => (
```
Comment (stevensJourney) [inline demos/diagnostics-app/src/app/views/layout.tsx:113]: I noticed that the items don't appear if the screen size is made smaller.
image

Comment (rkistner) [reply demos/diagnostics-app/src/app/views/layout.tsx:104]: Thanks, fixed.

File: demos/diagnostics-app/src/app/views/layout.tsx [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+import { useNavigationPanel } from '@/components/navigation/NavigationPanelContext';
+import { signOut, sync, syncErrorTracker } from '@/library/powersync/ConnectionManager';
+import { usePowerSync } from '@powersync/react';
+import { useNavigate } from 'react-router-dom';
+
+export default function ViewsLayout({ children }: { children: React.ReactNode }) {
+  const powerSync = usePowerSync();
+  const navigate = useNavigate();
+
+  const [syncStatus, setSyncStatus] = React.useState(sync.syncStatus);
```
[the remark this answers is not in the document]
Comment (rkistner) [reply demos/diagnostics-app/src/app/views/layout.tsx]: Done

File: demos/diagnostics-app/package.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "@journeyapps/powersync-diagnostics-app",
```
[the remark this answers is not in the document]
Comment (rkistner) [reply demos/diagnostics-app/package.json]: Done

**Review by stevensJourney — APPROVED** [review]
LGTM. This is a very cool tool.

**Review by rkistner — APPROVED** [review]

**Review by stevensJourney — APPROVED** [review]

**Pull request merged**